CRM/Extension/Browser.php

```diff
@@ -222,14 +222,14 @@ private function grabRemoteJson() {
       ]);
     }
     catch (GuzzleException $e) {
-      throw new CRM_Extension_Exception(ts('The CiviCRM public extensions directory at %1 could not be contacted - please check your webserver can make external HTTP requests', [1 => $this->getRepositoryUrl()]), 'connection_error');
+      throw new CRM_Extension_Exception(ts('It is not possible to contact the CiviCRM extensions directory. You may be missing out on the latest updates to extensions. Check that you can view the <a href="%1">extension feed</a>. If that works check that your webserver can make external HTTP requests.', [1 => $this->getRepositoryUrl()]), 'connection_error');
     }
     finally {
       restore_error_handler();
     }
 
     if ($response->getStatusCode() !== 200) {
-      throw new CRM_Extension_Exception(ts('The CiviCRM public extensions directory at %1 could not be contacted - please check your webserver can make external HTTP requests', [1 => $this->getRepositoryUrl()]), 'connection_error');
+      throw new CRM_Extension_Exception(ts('It is not possible to contact the CiviCRM extensions directory. You may be missing out on the latest updates to extensions. Check that you can view the <a href="%1">extension feed</a>. If that works check that your webserver can make external HTTP requests.', [1 => $this->getRepositoryUrl()]), 'connection_error');
```

Comment:
Just a technical thing on the ts() for links they are written as `<a %1>` and then the href is in the array, like:

`throw new CRM_Extension_Exception(ts('It is not possible to contact the CiviCRM extensions directory. You may be missing out on the latest updates to extensions. Check that you can view the <a %1>extension feed</a>. If that works check that your webserver can make external HTTP requests.', [1 => 'href="' . $this->getRepositoryUrl() . '"']), 'connection_error');` See https://docs.civicrm.org/dev/en/latest/translation/#best-practices where it says "Hyperlinks within larger blocks of text". Mostly it avoids having to retranslate if the params change.